### PR TITLE
Simple Checker for checking validity of output YAML stream

### PR DESCRIPTION
**Pull request by vijayphoenix**
Features added
1. A checker for the dumped YAML stream.
2. Also, show the differences detected at the Event level. (Helpful for testing)

To be Done:
1. Showing error/difference reports with the proper source location information at the Event level

### CODE REVIEW

File: src-test/Main.hs [hunk 1]
```diff
@@ -48,6 +48,12 @@ main = do
           hPutStrLn stderr "unexpected arguments passed to yaml2event sub-command"
           exitFailure
 
+    ("check":args')
```
Comment (hvr) [inline src-test/Main.hs]: let's call this `yaml2yaml-validate` instead; as it's yet another yaml-to-yaml mode

```suggestion
    ("yaml2yaml-validate":args')
```

File: src-test/Main.hs [hunk 2]
```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
...
+      case (newevstrm) of
+        Left (ofs',msg') -> do
+          case msg' of
+            "" -> hPutStrLn stderr ("Parsing error in the generated YAML stream near byte offset " ++ show ofs')
+            _  -> hPutStrLn stderr ("Parsing error in the generated YAML stream near byte offset " ++ show ofs' ++ " (" ++ msg' ++ ")")
+          exitFailure
+        Right newEvents -> do
+          let newEvLen = length newEvents
+          hPutStrLn stdout $ printf "\nInput  Event Stream Length: %d\nOutput Event Stream Length: %d\n" oldEvLen newEvLen
+          let diffList = filter (\(a,b)-> if(a/=b) then True else False) $ zip oldEvents newEvents
```
Comment (hvr) [inline src-test/Main.hs]:
```suggestion
          let diffList = filter (uncurry (/=)) $ zip oldEvents newEvents
```

Also note that

```hs
(if b then True else False) == b
```

File: src-test/Main.hs [hunk 3]
```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
...
+          hPutStrLn stdout $ printf "No of difference detected: %d\n" $ length diffList 
+          forM_ (diffList) $ \(old,new) -> do
```
Comment (hvr) [inline src-test/Main.hs]:
```suggestion
          forM_ diffList $ \(old,new) -> do 
```

File: src-test/Main.hs [hunk 4]
```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
...
     inYamlDat <- BS.L.getContents
     print (length (parseEvents' inYamlDat))
   where
     parseEvents' = map (either (\(ofs,msg) -> error ("parsing error near byte offset " ++ show ofs ++ " (" ++ msg ++ ")")) id) . parseEvents
 
+cmdCheck :: IO() 
+cmdCheck = do
+  inYamlDat <- BS.L.getContents
+  let oldevstrm = sequence $ parseEvents inYamlDat
+  case (oldevstrm) of
```
Comment (hvr) [inline src-test/Main.hs]: redundant brackets... and why not inline the definition of `oldevstrm`?

File: src-test/Main.hs [hunk 5]
```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
...
+    Left (ofs,msg) -> do
+      case msg of
+        "" -> hPutStrLn stderr ("Parsing error near byte offset " ++ show ofs)
+        _  -> hPutStrLn stderr ("Parsing error near byte offset " ++ show ofs ++ " (" ++ msg ++ ")")
+      exitFailure
+    Right oldEvents -> do
+      let oldEvLen = length oldEvents
+          output = (writeEvents YT.UTF8 oldEvents)
```
Comment (hvr) [inline src-test/Main.hs]: redundant brackets; oldEvLen can be inlined

File: src-test/Main.hs [hunk 6]
```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
...
+      BS.L.hPutStr stdout output
+      hFlush stdout
+      let newevstrm = sequence $ parseEvents output
+      case (newevstrm) of
```
Comment (hvr) [inline src-test/Main.hs]: brackets... also why not `case sequence (parseEvents output) of ...` ?

File: src-test/Main.hs [hunk 7]
```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
...
 
+cmdCheck :: IO() 
+cmdCheck = do
+  inYamlDat <- BS.L.getContents
+  let oldevstrm = sequence $ parseEvents inYamlDat
+  case (oldevstrm) of
+    Left (ofs,msg) -> do
+      case msg of
+        "" -> hPutStrLn stderr ("Parsing error near byte offset " ++ show ofs)
+        _  -> hPutStrLn stderr ("Parsing error near byte offset " ++ show ofs ++ " (" ++ msg ++ ")")
```
Comment (hvr) [inline src-test/Main.hs]: `case` can be turned into

```suggestion
        hPutStrLn stderr ("Parsing error near byte offset " ++ show ofs ++ (if null msg then "" else " (" ++ msg ++ ")"))
```

File: src-test/Main.hs [hunk 8]
```diff
@@ -159,14 +166,44 @@ cmdYaml2Event = do
...
+      let oldEvLen = length oldEvents
+          output = (writeEvents YT.UTF8 oldEvents)
+      BS.L.hPutStr stdout output
+      hFlush stdout
+      let newevstrm = sequence $ parseEvents output
+      case (newevstrm) of
+        Left (ofs',msg') -> do
+          case msg' of
+            "" -> hPutStrLn stderr ("Parsing error in the generated YAML stream near byte offset " ++ show ofs')
+            _  -> hPutStrLn stderr ("Parsing error in the generated YAML stream near byte offset " ++ show ofs' ++ " (" ++ msg' ++ ")")
```
Comment (hvr) [inline src-test/Main.hs]: see comment above about case/if transformation

File: src-test/Main.hs (hunk 5 above)
Comment (vijayphoenix) [reply src-test/Main.hs]: oldEvLen used later also

**Pull request merged**